packages/dd-trace/src/appsec/index.js

```diff
@@ -163,8 +165,9 @@ function incomingHttpEndTranslator ({ req, res }) {
     persistent[addresses.HTTP_INCOMING_COOKIES] = req.cookies
   }
 
-  if (req.query !== null && typeof req.query === 'object') {
-    persistent[addresses.HTTP_INCOMING_QUERY] = req.query
+  const query = req.query
+  if (query !== null && typeof query === 'object') {
+    persistent[addresses.HTTP_INCOMING_QUERY] = query
   }
```

Comment:
wait why is query still there since we instrument it anyway ?

Reply:
For the same reason why we are doing it for `req.query` and `req.cookies`? Do you think that we should delete this code? I think that it could be usefull only for frameworks that are not express.

Reply:
body and cookies are still in this subscriber because there is a million different body and cookie parser, and we only support 1, so in case there also here we send them to the WAF. But for querystring, there is only one parser no ? it's directly built into express.
We should not delete either body or cookie, but query yes

Reply:
@uurien 
> I think that it could be usefull only for frameworks that are not express.

This code was never designed for frameworks other than express. What other stupid framework would mutate the req object directly :laughing:

Reply:
@simon-id  I thought that the only used framework for query and cookies is only express not all of them

Reply:
no that is true, we only support express, but express can have many different kind of body-parser or cookie-parser middlewares, so just in case, we read it at the end of the request too

Reply:
This part is needed for nextjs, PR to add explanation https://github.com/DataDog/dd-trace-js/pull/4967/files